eng/pipelines/official.yml

```diff
@@ -133,10 +140,28 @@ extends:
             deploy:
               steps:
               - task: 1ES.DownloadPipelineArtifact@1
-                displayName: 🟣 Import PackageArtifacts from Previous Stage
+                displayName: 🟣 Download build artifacts
                 inputs:
-                  artifactName: Artifacts_Windows_NT_Release
-                  targetPath: $(Pipeline.Workspace)/Artifacts_Windows_NT_Release
-              - template: /eng/pipelines/templates/steps/workload-publish.yml@self
-                parameters:
-                  feedForPublishing: ${{ parameters.feedForPublishing }}
+                  artifactName: Artifacts
+                  targetPath: $(Build.SourcesDirectory)/artifacts
+              # 1ES docs: https://eng.ms/docs/cloud-ai-platform/devdiv/one-engineering-system-1es/1es-docs/1es-pipeline-templates/features/outputs/nuget-packages
+              # DotNetCoreCLI@2 docs: https://learn.microsoft.com/en-us/azure/devops/pipelines/tasks/reference/dotnet-core-cli-v2
+              - ${{ if eq(parameters.publishToAzDO, true) }}:
+                - task: 1ES.PublishNuget@1
+                  displayName: 🟣 Publish packages to AzDO
+                  inputs:
+                    useDotNetTask: true
+                    packagesToPush: $(Build.SourcesDirectory)/artifacts/packages/**/*.nupkg
+                    packageParentPath: $(Build.SourcesDirectory)/artifacts
+                    publishVstsFeed: ${{ parameters.feedForPublishing }}
+              - ${{ if eq(parameters.publishToNuGet, true) }}:
+                - task: 1ES.PublishNuget@1
+                  displayName: 🟣 Publish packages to NuGet.org
+                  inputs:
+                    useDotNetTask: false
```

Comment:
Trying it for now with NuGet CLI instead of DotNet CLI. I don't know if it'll make a difference or not.